app/src/main/java/com/paranoidandroid/journey/wizard/fragments/LegsFragment.java

```diff
@@ -123,6 +124,17 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
 
     }
 
+    private String generateDisplayName(String destination) {
+        String[] components = destination.split(",");
+        if (components.length == 2) {
+            return destination;
+        } else if (components[2].equals(" United States")) {
+            return components[0] + "," + components[1];
+        } else {
+            return components[0] + "," + components[components.length -1];
+        }
```

Comment:
Also in a next PR we should make basic index checking here, i.e.
`else if (components.length > 2 && components[2].equals(" United States"))` etc